kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/producer/KafkaProducerService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.homeoffice.digital.sas.kafka.producer;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+
+@Component
+@EnableAutoConfiguration
+public class KafkaProducerService<T> {
+
+  private final KafkaTemplate<String, KafkaEventMessage<T>> kafkaTemplate;
+  private final String topicName;
+  private final String projectVersion;
+
+  public KafkaProducerService(
+      KafkaTemplate<String, KafkaEventMessage<T>> kafkaTemplate,
+      @Value("${spring.kafka.template.default-topic}") String topicName,
```

Comment:
I know this is lift and shift from timecard but I'm not sure this is the best way to set up the topic name. This means that one service will only ever be able to write to one topic.

Reply:
Happy with a separate task, thanks

Reply:
Task created: https://collaboration.homeoffice.gov.uk/jira/browse/EAHW-2639